target/linux/realtek/image/rtl838x.mk

```diff
@@ -39,6 +39,14 @@ TARGET_DEVICES += d-link_dgs-1210-20
 define Device/d-link_dgs-1210-28
   $(Device/d-link_dgs-1210)
   DEVICE_MODEL := DGS-1210-28
+  DLINK_KERNEL_PART_SIZE := 1572864
+  CAMEO_KERNEL_PART := 2
+  CAMEO_ROOTFS_PART := 3
+  CAMEO_CUSTOMER_SIGNATURE := 2
+  CAMEO_BOARD_VERSION := 32
+  IMAGES += factory_only_for_image1.bin
```

Comment:
Deviations from just the `factory.bin` name are normally used to support (device or firmware) revisions that require different factory images, not to prevent users from shooting themselves in the foot. Some users can (and probably will) still just ignore the file name and flash to image2.

Could you also document how to debrick the device in the commit message, in case an image is flashed to image2? It appears this device was added without a proper description, so it would be good to have this info.

Reply:
> Deviations from just the `factory.bin` name are normally used to support (device or firmware) revisions that require different factory images, not to prevent users from shooting themselves in the foot. Some users can (and probably will) still just ignore the file name and flash to image2.

I know but we always try hard to avoid it.
 
> Could you also document how to debrick the device in the commit message, in case an image is flashed to image2? It appears this device was added without a proper description, so it would be good to have this info.

It's complicated. The device does have an emergency recovery from u-boot, but we can only trigger it from a working original image. It looks like the reset button does not work from a cold reboot. It is not just for this recovery: not even config reset works from a cold reboot (or from an OpenWrt boot). The only recovery possible is by triggering it from a serial console by pressing ESC. From there, there easiest way to recovery is using the D-Link Network Assistant v2.0.2.4. If you want to do it manually, you'll need to cut the firmware where the kernel blob+cameo tag ends, remove header from both parts and flash each one to the corresponding partition. I'll add these info to the commit message (and wiki).

> I'm not entirely convinced about the `factory_only_for_image1.bin` name, but maybe `factory_image1.bin` would be an option.

I'm fine with factory_image1.bin, if it will be accepted.